sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -135,6 +135,9 @@ def __init__(self, check):
         self.dm_exec_query_stats_row_limit = int(
             check.statement_metrics_config.get('dm_exec_query_stats_row_limit', 10000)
         )
+        self.enforce_collection_interval_deadline = bool(
+            check.statement_metrics_config.get('enforce_collection_interval_deadline', True)
```

Comment:
Is this config option being added in a different PR?

Reply:
This option is not meant to be user facing. It's mostly so that we can disable the deadline during integration tests. 

I can't think of a reason currently why it would be need to be disabled in production but it's also good to have the option there in case it does become necessary for some customer while troubleshooting.

Reply:
I think it would be helpful to still document it in `spec.yaml` but just add `hidden: true` like for our [openmetrics_legacy](https://github.com/DataDog/integrations-core/blob/master/openmetrics/assets/configuration/spec.yaml#L17) template. This will hide it from the `conf.yaml.example` file but it's still documented.

Reply:
Oh cool didn't realize there was a hidden option. Updated